patroni/api.py

```diff
@@ -677,9 +1106,39 @@ def query(self, sql, *params, **kwargs):
     def get_postgresql_status(self, retry: Optional[bool] = False) -> Dict[str, Any]:
         """Builds an object representing a status of "postgres".
 
-        Some of values are collected by executing a query and other are taken from the state stored in memory.
+        Some of the values are collected by executing a query and other are taken from the state stored in memory.
+
         :param retry: whether the query should be retried if failed or give up immediately
-        :returns: a dict with the status of Postgres/Patroni
+        :returns: a dict with the status of Postgres/Patroni. The keys are:
+            * ``state``: Postgres state among ``stopping``, ``stopped``, ``stop failed``, ``crashed``, ``running``,
+                ``starting``, ``start failed``, ``restarting``, ``restart failed``, ``initializing new cluster``,
+                ``initdb failed``, ``running custom bootstrap script``, ``custom bootstrap failed``,
+                ``creating replica``, or ``unknown``;
+            * ``postmaster_start_time``: ``pg_postmaster_start_time()``;
+            * ``role``: ``replica`` or ``master`` based on ``pg_is_in_recovery()`` output;
+            * ``server_version``: Postgres version without periods, e.g. ``150002`` for Postgres ``15.2``;
+            * ``xlog``: dictionary. Its structure depends on ``role``:
+                * If ``master``:
+                    * ``location``: ``pg_current_wal_lsn()``
+                * If ``replica``:
+                    * ``received_location``: ``pg_wal_lsn_diff(pg_last_wal_receive_lsn(), '0/0')``;
+                    * ``replayed_location``: ``pg_wal_lsn_diff(pg_last_wal_replay_lsn(), '0/0)``;
+                    * ``replayed_timestamp``: ``pg_last_xact_replay_timestamp``;
+                    * ``paused``: ``pg_is_wal_replay_paused()``;
+            * ``sync_standby``: ``True`` if replication mode is synchronous and this is a sync standby;
+            * ``timeline``: PostgreSQL primary node timeline;
+            * ``replication``: :class:`list` of :class:`dict` entries, one for each replication connection. Each entry
+                contains the following keys:
+                * ``application_name``: ``pg_stat_get_activity().application_name``;
+                * ``client_addr``: ``pg_stat_get_activity().client_addr``;
+                * ``state``: ``pg_stat_get_wal_senders().state``;
+                * ``sync_priority``: ``pg_stat_get_wal_senders().sync_priority``;
+                * ``sync_state``: ``pg_stat_get_wal_senders().sync_state``;
+                * ``usename``: name of the role ``pg_stat_get_activity().usesysid``.
```

Comment:
I think here we could safely refer to columns in pg_stat_replication view here.
The view isn't used directly only for "performance" reasons to avoid unnecessary joins.
Under the hood the view is calling exactly these functions.

Reply:
Makes sense!